src/aiokatcp/sensor.py

```diff
@@ -162,14 +169,16 @@ def __init__(
         self.auto_strategy_parameters = tuple(auto_strategy_parameters)
         # TODO: should validate the parameters against the strategy.
 
-    def notify(self, reading: Reading[_T]) -> None:
+    def notify(self, reading: Reading[_T], old_reading: Reading[_T]) -> None:
         """Notify all observers of changes to this sensor.
 
         Users should not usually call this directly. It is called automatically
         by :meth:`set_value`.
         """
-        for observer in self._observers:
-            observer(self, reading)
+        for classic_observer in self._classic_observers:
+            classic_observer(self, reading)
+        for change_aware_observer in self._change_aware_observers:
+            change_aware_observer(self, reading, old_reading)
```

Comment:
Since you're identifying the type of observer by the name rather than the position of the argument, I'd rather pass the argument as a kwarg. Note that `Callable` can't express kwargs, so you have to define a `Protocol` with ` __call__` method.